### PR TITLE
refactor: move release notes to be version agnostic

### LINKED ISSUE #871: Release notes should be version agnostic
Currently, we host release notes under `docs/v0.7/en/release-notes`.

This is not scalable, because we should not copy the whole release note folder to `docs/v0.8/en` when v0.8 is cut.

Instead, we should make the release notes version agnostic. Here are two possible ways:

1. Lift the release-notes folder and set symbol link at each version directory, use the `summary.yml` to control which range of release notes should be displayed.

This is easy to implement, but our pre-commit is unhappy with this:

```
> greptime_docs@1.0.0 lint-staged /Users/tison/GreptimeWorkspace/docs
> npx lint-staged

✖ error: 'docs/v0.7/en/release-notes/all-releases.md' is beyond a symbolic link
  fatal: Unable to process path docs/v0.7/en/release-notes/all-releases.md
  Cannot save the current worktree state
```

Related to https://github.com/lint-staged/lint-staged/issues/1318.

2. Copy the files on build, or use branches for each version.

This is how [pingcap/docs](https://github.com/pingcap/docs/tree/master) works - one branch for a version. It may require more complex build set up.

What do you think? @GreptimeTeam/marketing @nicecui @Aganivi @alili

### PR DESCRIPTION
**Pull request by tisonkun**
## What's Changed in this PR

This closes #871 

## Checklist

- [ ] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/auto-imports.d.ts
Comment (tisonkun) [inline docs/auto-imports.d.ts:1]: Should be reverted. Processing ...

**Review by alili — APPROVED** [review]
LGTM

**Review by Aganivi — COMMENTED** [review]
It can't display release-notes in the previous version, and Changelog was lost in lastest version. Please check it. https://symlink.greptime-docs.pages.dev/v0.6/

**Comment by tisonkun** [comment]
@Aganivi This is expected.

**Comment by tisonkun** [comment]
For further improvment, we can modify `getSrcExclude` method and exclude release notes >= selected version.

**Review by Aganivi — APPROVED** [review]

**Pull request merged**